### PR TITLE
[#15] [UI] As a signed-in user, I can see the Home Screen

### LINKED ISSUE #15: [UI] As a signed-in user, I can see the Home Screen
## Why

* Allow user to see Home screen after signing in

## Acceptance Criteria

* Show the header
* Show current date
* Show user's avatar
* Show navigation indicator
* Show survey card
 
* Show survey's cover image
* Show survey's title
* Show survey's description
* Show the next circle button
* User can swipe left or right to see more surveys

## Design

Screen Shot 2022-05-09 at 10 53 04

### PR DESCRIPTION
**Pull request by thiennguyen0196**
- Close #15 

## What happened 👀

- Create home screen UI
- Integrate calling get survey for first page to display data
- Add library smooth page indicator to use its UI for the circle indicator
- Convert `HomeScreen` from `StatelessWidget` to `ConsumerStatefulWidget`

## Insight 📝

- The unit test will be added later in #17 and #18 

## Proof Of Work 📹

- Android

[device-2023-04-28-170711.webm](https://user-images.githubusercontent.com/25218255/235119883-832a337e-4843-4264-87dd-c84d2e40997f.webm)

- iOS

https://user-images.githubusercontent.com/25218255/235119875-1746da88-c418-4345-ba64-4aee4a5ebdaa.mp4

### CODE REVIEW

File: lib/ui/home/home_screen.dart [hunk 1]
```diff
@@ -1,29 +1,176 @@
...
+      });
+
+  Widget _pageIndicatorWidget(BuildContext context, int length) =>
+      SmoothPageIndicator(
+        controller: _pageController,
+        count: length,
+        effect: const ScrollingDotsEffect(
+          dotColor: Colors.white24,
+          activeDotColor: Colors.white,
+          dotHeight: 8,
```
Comment (ducbm051291) [inline lib/ui/home/home_screen.dart:97]: Same here

**Review by ducbm051291 — APPROVED** [review]
Apart from the decimal number and conflict need to rebase. The rest are LGTM.

File: lib/ui/home/home_header_widget.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  const HomeHeaderWidget({
+    super.key,
+    required this.profileImgUrl,
+    required this.profileImgClickCallback,
+  });
+
+  String get _todayDate {
+    final today = DateTime.now();
+    return '${DateFormat.EEEE().format(today)}, ${DateFormat.MMMMd().format(today)}'
```
Comment (nmint8m) [inline lib/ui/home/home_header_widget.dart:21]: It's better to have a `DateFormatter` class. What do you think?

File: lib/ui/home/home_view_model.dart [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+}
+
+class LoadMoreDataSet {
+  int page = _pageDefault;
+  int pageSize = _pageSizeDefault;
+  bool isHasMore = true;
+  bool isLoading = false;
+}
```
Comment (nmint8m) [inline lib/ui/home/home_view_model.dart:76]: This class is only used in this file, please consider to mark it as private class. `_LoadMoreDataSet`

**Review by nmint8m — COMMENTED** [review]
Please take a look 😄

**Comment by nmint8m** [comment]
Rest looks good to me. Please resolve the conflicts.

**Comment by thiennguyen0196** [comment]
The conflict and all discussion are resolve, please help me to review it @nmint8m 😄

File: lib/ui/home/home_footer_widget.dart [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        ),
+        const SizedBox(height: Dimensions.paddingNormal),
+        Row(
+          children: [
+            Expanded(
+                child: Text(
+              _survey?.description ?? '',
+              style: context.textTheme.bodyMedium,
+              maxLines: 2,
+            )),
```
Comment (nmint8m) [inline lib/ui/home/home_footer_widget.dart:30]: Please add `,` here.
```suggestion
            ),
          ),
```

File: lib/ui/home/home_screen.dart [hunk 5]
```diff
@@ -1,29 +1,176 @@
...
+        effect: const ScrollingDotsEffect(
+          dotColor: Colors.white24,
+          activeDotColor: Colors.white,
+          dotHeight: 8.0,
+          dotWidth: 8.0,
+        ),
+      );
+
+  Widget _homeWidget(BuildContext context) {
+    GlobalKey<ScaffoldState> scaffoldKey = GlobalKey();
```
Comment (nmint8m) [inline lib/ui/home/home_screen.dart:103]: Let's define it as a global variable.

File: lib/ui/home/home_screen.dart [hunk 6]
```diff
@@ -1,29 +1,176 @@
...
+  Widget _homeHeaderWidget({
+    required BuildContext context,
+    required VoidCallback openSideMenuCallback,
+  }) {
+    return HomeHeaderWidget(
+      profileImgUrl:
+          'https://secure.gravatar.com/avatar/6733d09432e89459dba795de8312ac2d',
+      profileImgClickCallback: openSideMenuCallback,
+    );
+  }
```
Comment (nmint8m) [inline lib/ui/home/home_screen.dart:82]: As we adopt  `ConsumerState` for `HomeScreenState`, so no need to define `context` as parameter, as `context` now is an instance variable of `HomeScreenState`.

Also I don't think we need to define `openSideMenuCallback` here. Just simply call the function:

```suggestion
  Widget _homeHeaderWidget() => HomeHeaderWidget(
      profileImgUrl: 'https://secure.gravatar.com/avatar/6733d09432e89459dba795de8312ac2d',
      profileImgClickCallback: () => scaffoldKey.currentState?.openEndDrawer(),
    );
```

**Review by nmint8m — APPROVED** [review]
Just small comments. Good to go. 🏃

**Pull request merged**